Stratis.Bitcoin/Utilities/LinqExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Linq
+{
+    public static class LinqExtensions
+    {
+        public static long Median(this IEnumerable<long> source)
```

Comment:
Maybe safer to use IEnumerable<TimeOffset> it will be more clear to other devs of the purpose of the mthod